addons/account/models/account_bank_statement.py

```diff
@@ -164,12 +108,43 @@ def _compute_date_index(self):
             stmt.first_line_index = sorted_lines[:1].internal_index
             stmt.date = sorted_lines.filtered(lambda l: l.state == 'posted')[-1:].date
 
-    @api.depends('balance_start', 'line_ids.amount', 'line_ids.state')
+    @api.depends('create_date')
+    def _compute_balance_start(self):
+        for stmt in self.sorted(lambda x: x.first_line_index or '0'):
+            previous_line_with_statement = self.env['account.bank.statement.line'].search([
+                ('internal_index', '<', stmt.first_line_index),
+                ('journal_id', '=', stmt.journal_id.id),
+                ('state', '=', 'posted'),
+                ('statement_id', '!=', False),
+            ], limit=1)
+            balance_start = previous_line_with_statement.statement_id.balance_end_real
+
+            lines_in_between_domain = [
+                ('internal_index', '<', stmt.first_line_index),
+                ('journal_id', '=', stmt.journal_id.id),
+                ('state', '=', 'posted'),
+            ]
+            if previous_line_with_statement:
+                lines_in_between_domain.append(('internal_index', '>', previous_line_with_statement.internal_index))
+                # remove lines from previous statement (when multi-editing a line already in another statement)
+                balance_start -= sum(previous_line_with_statement.statement_id.line_ids.filtered(
+                    lambda l: l.id in stmt.line_ids._origin.ids).mapped('amount'))
```

Comment:
cosmetic suggestion:
```suggestion
                previous_st_lines = previous_line_with_statement.statement_id.line_ids
                lines_in_common = previous_st_lines.filtered(lambda l: l.id in stmt.line_ids._origin.ids
                balance_start -= sum(lines_in_common.mapped('amount'))
```